plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetInstallDelegate.java

```diff
@@ -118,4 +132,54 @@ private static void createConfigFiles(IProject project, IProgressMonitor monitor
         configFileLocation, AppEngineTemplateUtility.APPENGINE_WEB_XML_TEMPLATE, Collections.<String, String> emptyMap());
   }
 
+  private static void addAppEngineJarsToMavenProject(IProject project, IProgressMonitor monitor) throws CoreException {
+    IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().getProject(project);
+    Model pom = facade.getMavenProject(monitor).getModel();
+
+    List<Dependency> currentDependencies = pom.getDependencies();
+    List<Dependency> updatedDependencies = updateMavenProjectDependencies(currentDependencies);
+    pom.setDependencies(updatedDependencies);
+    updatePomProperties(pom.getProperties(), monitor);
+
+    DefaultModelWriter writer = new DefaultModelWriter();
+    File pomFile = pom.getPomFile();
+    try {
+      writer.write(pomFile, null /* options */, pom);
+    } catch (IOException e) {
+      throw new CoreException(StatusUtil.error(FacetInstallDelegate.class, e.getMessage()));
+    }
+    project.getFile(pomFile.getName()).refreshLocal(IResource.DEPTH_ZERO, monitor);
+  }
+
+  // visible for testing
+  public static List<Dependency> updateMavenProjectDependencies(List<Dependency> initialDependencies) {
+    List<Dependency> allAppEngineDependencies = MavenAppEngineFacetUtil.getAppEngineDependencies();
+    if (initialDependencies == null) {
+      return allAppEngineDependencies;
+    }
+
+    if (initialDependencies.size() == 0) {
+      return allAppEngineDependencies;
+    }
+
+    for (Dependency appEngineDependency : allAppEngineDependencies) {
+      if(!MavenAppEngineFacetUtil.doesListContainDependency(initialDependencies, appEngineDependency)) {
+        initialDependencies.add(appEngineDependency);
+      }
+    }
+
+    return initialDependencies;
+  }
+
+  //visible for testing
+  public static void updatePomProperties(Properties projectProperties, IProgressMonitor monitor) {
```

Comment:
https://github.com/GoogleCloudPlatform/cloud-tools-for-eclipse/pull/627/files#r77026534